src/FileCache.js

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2017 Electric Imp
+// This file is licensed under the MIT License
+// http://opensource.org/licenses/MIT
+
+'use strict';
+
+const fs = require('fs-extra');
+const url = require('url');
+const path = require('path');
+const minimatch = require('minimatch');
+const HttpReader = require('./Readers/HttpReader');
+const GithubReader = require('./Readers/GithubReader');
+const GITHUB_DIR = path.sep + 'github';
```

Comment:
I still believe we shouldn't have two separate folders for github and http resources.